backend/src/interface/persistent_data/model/mod.rs

```diff
@@ -0,0 +1,125 @@
+//! These structs are used for database operations
+use crate::util::{self, Date};
+use crate::util::{Additive, Allergen, Price};
+use std::error::Error;
+
+use std::string::String;
+use thiserror::Error;
+use util::MealType;
+use uuid::Uuid;
+
+/// Enumerations for possible data request faults
+#[derive(Debug, Error)]
+pub enum DataError {
```

Comment:
Maybe put the error inside the `persistent_datamod.rs` directly, as it is not part of the model?

Reply:
I can't see the advantage of that. The idea behind this structure: Interfaces into root, Structs/Enums into root/model.

Reply:
For me, model are just data models. The error type has a more universal role imo.